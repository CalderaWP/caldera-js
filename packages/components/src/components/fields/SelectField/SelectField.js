import PropTypes from 'prop-types';
import React from 'react';
import { SelectControl } from '@wordpress/components';
import { parseAttributes, fieldClassNames, isValidHtml5type } from '../util';
export const SelectField = ({
	attributes,
	label,
	fieldId,
	onChange,
	value,
	options,
	multiple,
	description,
	placeholder
}) => {
	attributes = parseAttributes(attributes, 'select');

	if (!value && placeholder) {
		options.unshift({
			label: placeholder
		});
	}
	return (
		<SelectControl
			className={fieldClassNames('select')}
			id={fieldId}
			value={value}
			options={options}
			onChange={onChange}
			label={label}
			help={description}
			{...attributes}
		/>
	);
};

SelectField.propTypes = {
	label: PropTypes.string,
	description: PropTypes.string,
	fieldId: PropTypes.string,
	required: PropTypes.bool,
	multiple: PropTypes.bool
};

SelectField.defaultProps = {
	required: false,
	multiple: false,
	description: ''
};
