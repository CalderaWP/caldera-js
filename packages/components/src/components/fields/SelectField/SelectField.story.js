import React from 'react';
import { storiesOf } from '@storybook/react';
import { SelectField } from './SelectField';

const onChange = newValue => {
	console.log(newValue);
};
const onBlur = event => {};

const STORY_NAME_SELECT = 'SelectField';
storiesOf(STORY_NAME_SELECT, module).add(
	'With no value selected and  placeholder',
	() => (
		<SelectField
			label={'Select A Hat'}
			onChange={onChange}
			description={'selection of hats'}
			fieldId={'selection-hats'}
			placeholder={'---'}
			required={true}
			options={[{ value: 1, label: 'One' }, { value: 2, label: 'Two' }]}
		/>
	)
);
storiesOf(STORY_NAME_SELECT, module).add(
	'With no value selected and no placeholder',
	() => (
		<SelectField
			label={'Select A Hat'}
			onChange={onChange}
			description={'selection of hats'}
			fieldId={'selection-hats'}
			required={true}
			options={[{ value: 1, label: 'One' }, { value: 2, label: 'Two' }]}
		/>
	)
);
storiesOf(STORY_NAME_SELECT, module).add('With value selected', () => (
	<SelectField
		label={'Select A Hat'}
		onChange={onChange}
		description={'selection of hats'}
		fieldId={'selection-hats'}
		required={true}
		value={2}
		options={[{ value: 1, label: 'One' }, { value: 2, label: 'Two' }]}
	/>
));
