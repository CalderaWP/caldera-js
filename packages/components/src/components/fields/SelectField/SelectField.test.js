import * as React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { SelectField } from './SelectField';

describe('Select Field component', () => {
	let onChange;
	beforeEach(() => {
		onChange = jest.fn();
	});
	it('matches snapshot, not multiple', () => {
		const component = renderer.create(
			<SelectField
				label={'Select A Hat!'}
				description={'selection of hats'}
				fieldId={'selection-hats'}
				required={false}
				multiple={false}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Fires on change event', () => {
		const component = mount(
			<SelectField
				label={'Select A Hat'}
				onChange={onChange}
				description={'selection of hats'}
				fieldId={'selection-hats'}
				required={true}
				multiple={true}
				options={[
					{ value: 1, label: 'One' },
					{ value: 2, label: 'Two' }
				]}
			/>
		);

		const event = { target: { value: 2 } };
		component.find('select').simulate('change', event);
		expect(onChange.mock.calls.length).toBe(1);
		expect(onChange.mock.calls[0][0]).toBe(2);
	});

	it('With no value selected and placeholder', () => {
		const placeholder = '---';
		const component = mount(
			<SelectField
				label={'Select A Hat'}
				onChange={onChange}
				description={'selection of hats'}
				fieldId={'selection-hats'}
				placeholder={placeholder}
				required={true}
				options={[
					{ value: 1, label: 'One' },
					{ value: 2, label: 'Two' }
				]}
			/>
		);
		expect(
			component
				.find('select')
				.find('option')
				.first()
				.props()
		).toEqual({ children: '---', value: undefined });
	});
});
