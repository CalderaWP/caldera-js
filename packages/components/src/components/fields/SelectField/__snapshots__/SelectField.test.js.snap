// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Field component matches snapshot, not multiple 1`] = `null`;
